tests/FluentAssertions.AspNetCore.Mvc.Tests/PhysicalFileResultAssertions_Tests.cs

```diff
@@ -1,50 +1,137 @@
-﻿using FluentAssertions.AspNetCore.Mvc.Tests.Helpers;
+﻿using System;
 using Microsoft.AspNetCore.Mvc;
-using System;
+using Microsoft.Net.Http.Headers;
 using Xunit;
 
 namespace FluentAssertions.AspNetCore.Mvc.Tests
 {
     public class PhysicalFileResultAssertions_Tests
     {
+        private const string TestPhysicalPath = "TestPhysicalPath";
+        private const string TestContentType = "text/html";
+        private const string TestFileDownloadName = "TestFileDownloadName";
+        private readonly DateTimeOffset? TestLastModified = DateTimeOffset.Now;
+        private readonly EntityTagHeaderValue TestEntityTag = new EntityTagHeaderValue("\"0815\"");
 
         [Fact]
-        public void WithFileName_GivenExpectedValue_ShouldPass()
+        public void WithPhysicalPath_GivenExpected_ShouldPass()
         {
-            var actualFileName = "Test1.txt";
-            var expectedFileName = string.Copy(actualFileName);
-            ActionResult result = TestDataGenerator.CreatePhysicalFileResult(actualFileName);
+            ActionResult result = new PhysicalFileResult(TestPhysicalPath, TestContentType);
+            result.Should().BePhysicalFileResult().WithPhysicalPath(TestPhysicalPath);
+        }
```

Comment:
I see you changed some of my test also. I'm interested in your way of testing.
I also share here my thoughts on testing:
When I'm testing and create lot of similar object I extract the creation into a TestDataGenerator and it creates the class with defaults. In the individual tests I just define the important values.
Also in this first test I used string.Copy(actualFileName) so it is a different instance. I did this because there should be lot more case that should be tested even in this simple case and sometimes I try to sneak in additional less important case with right choosing of parameters. This time I was testing if not just reference equality test goes under the hood.

Reply:
Hi @faddiv 
thanks for your feedback.
In general we can distinguish between 2 types of tests.
- Unit Tests
- Integration (EndToEnd) Tests.

The Code above is a unit Test. Its purpose is only to test our Assertions Code.
We want to demonstrate that the WithPhysicalPath Method works as expected.
And this Test shall demonstrate that it does not throw an assertion exception if the Path is correct.
Your code above does to much, because it also tests the behavoir of the .Net Framework itself. Hence it tests the internal functionality of the PhysicalFileResult.
Therefore it is more like an Integration Test.
I have a separate Sample Project, where I have Integration Tests.

Furthermore you don't need string.Copy. Use string constants in your test as well.
It is a basic rule to avoid magic strings (and numbers).
If I need a constant in more than one Test I use a private constant in the class.

Reply:
Yeah you are right. Sometimes I forget these tests are whitebox tests I shouldn't worry about the possible implementation details. Also most of the times I also avoid magic strings. In this case I just tried to follow the pattern used by others in this project.